src/Microsoft.ML.Recommender/RecommenderCatalog.cs

```diff
@@ -55,11 +55,24 @@ internal RecommendationTrainers(RecommendationCatalog catalog)
             /// <param name="matrixColumnIndexColumnName">The name of the column hosting the matrix's column IDs.</param>
             /// <param name="matrixRowIndexColumnName">The name of the column hosting the matrix's row IDs.</param>
             /// <param name="labelColumn">The name of the label column.</param>
+            /// <param name="approximationRank">Rank of approximation matrixes.</param>
+            /// <param name="learningRate">Initial learning rate. It specifies the speed of the training algorithm.</param>
+            /// <param name="numIterations">Number of training iterations.</param>
+            /// <example>
+            /// <format type="text/markdown">
+            /// <![CDATA[
+            ///  [!code-csharp[MatrixFactorization](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Recommendation/MatrixFactorization.cs)]
+            /// ]]></format>
+            /// </example>
             public MatrixFactorizationTrainer MatrixFactorization(
```

Comment:
>MatrixFactorization [](start = 46, length = 19)

Loss function is an important parameter. Could we have it here?

Reply:
as soon as I add loss function I need to add rest of option parameters.
We have another overload with options, if user want to specify loss functions, he can use it.

---
In reply to: [254913045](https://github.com/dotnet/machinelearning/pull/2451#discussion_r254913045) [](ancestors = 254913045)